safehttp/plugins/cors/cors.go

```diff
@@ -0,0 +1,215 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cors
+
+import (
+	"net/textproto"
+	"strconv"
+	"strings"
+
+	"github.com/google/go-safeweb/safehttp"
+)
+
+const requiredHeader string = "X-Cors"
+
+var disallowedContentTypes = map[string]bool{
+	"application/x-www-form-urlencoded": true,
+	"multipart/form-data":               true,
+	"text/plain":                        true,
+}
+
+// Interceptor handles CORS requests based on its settings.
+//
+// For more info about CORS, see: https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS
+//
+// Constraints
+//
+// The content types "application/x-www-form-urlencoded", "multipart/form-data"
+// and "text/plain" are banned and will result in a  415 Unsupported Media Type
+// response.
+//
+// Each CORS request must contain the header "X-Cors: 1".
+//
+// All of this is to prevent XSRF.
+type Interceptor struct {
+	// AllowedOrigins determines which origins should be allowed in the
+	// Access-Control-Allow-Origin header.
+	AllowedOrigins map[string]bool
+	allowedHeaders map[string]bool
+	// ExposedHeaders determines which headers should be set in the
+	// Access-Control-Expose-Headers header. This controls which headers in the
+	// response should be accessible by JavaScript.
+	//
+	// If ExposedHeaders is nil, then the header is not set, meaning that nothing
+	// is exposed.
+	ExposedHeaders []string
+	// AllowCredentials determines if Access-Control-Allow-Credentials should be
+	// set to true, which would allow cookies to be attached to requests.
+	AllowCredentials bool
+	// MaxAge sets the Access-Control-Max-Age header. MaxAge indicates how long,
+	// in seconds, that the results of a preflight request can be cached.
+	// MaxAge=0 results in MaxAge: 5.
+	// This default of 5 seconds is set because it is the default used by Chromium
+	// according to https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Max-Age
```

Comment:
```suggestion
	// MaxAge sets the Access-Control-Max-Age header, indicating how many seconds the results of a preflight
	// request can be cached.
	// Max Age=0 results in MaxAge:5,  the default used by Chromium
	// according to https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Max-Age
```
or something along those lines